jobQueue.py

```diff
@@ -317,3 +303,50 @@ def reset(self):
         """
         self.liveJobs._clean()
         self.deadJobs._clean()
+        self.unassignedJobs._clean()
+
+
+    def getNextPendingJob(self):
+        """Gets the next unassigned live job. Note that this is a 
+           blocking function and we will block till there is an available 
+           job.
+        """
+        # Blocks till the next item is added 
+        id = self.unassignedJobs.get()
```

Comment:
This is a really good addition - previously jobManager would just keep spinning (with a small delay)